### DISCUSSION TITLE
Latex - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#102] [Announcements]
# Latex - CS自学指南

CS自学指南

https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/Latex/

**Comment by cxchhh** [comment]
LaTeX和markdown适用范围有什么不同？

**Reply by VortexBlade** [reply]
Markdown编辑器一般都支持LaTeX吧

**Reply by yliu-fe** [reply]
Markdown可以写github的readme和静态网页，但latex不能（
LaTeX的构造相对复杂，模板、图文、公式、页眉页脚，都要设置，毕竟是一种直接编译PDF的语言
Markdown就比较简单，章节、粗体、斜体什么的都比较方便，坏处是导出成PDF就不美观。此外，用过Jupyter就会发现Markdown可以拿来在代码块之间写笔记，非常方便。
但两套系统共用一套数学符号的表达方法
所以正式文档和PPT（Beamer）我用LaTeX写，日常笔记就写Markdown

**Comment by CHENLexiao8848** [comment]
一般mac使用什么latex软件比较好呀

**Reply by LysanderT** [reply]
可以用vsc配置latex

**Reply by LilacZhang** [reply]
也可以用sublime text，zhihu上有相关的配置教程

**Comment by daijun10086** [comment]
overleaf是真的神器！！

**Comment by Araik-77** [comment]
数学课入LaTex,用哪门数学课啊

**Reply by LyshmilyY** [reply]
线代

**Comment by sandyzikun** [comment]
算是我個人在`LaTeX`上的一點折騰的經驗, 望能夠與大家分享交流... 本地配置`LaTeX`環境主流的方式是[`TeX Live`](https://mirrors.ustc.edu.cn/CTAN/systems/texlive/)與[`MiKTeX`](https://mirrors.ustc.edu.cn/CTAN/systems/win32/miktex/), 後者自帶`TeXworks`編輯器, 遇到沒有安裝的包也可以自動安裝, 可以説是開箱即用, 而且這兩者都是跨平臺的. 這些資源基本上都可以在[`CTAN`](https://ctan.org/)及其鏡像(如[科大鏡像](https://mirrors.ustc.edu.cn/CTAN/))上找到.

**Reply by sandyzikun** [reply]
此外`overleaf`是提供社區開源版本的, 有服務器或者虛擬機的朋友可以自行搭建, git-repo位於 <https://github.com/overleaf/toolkit.git>, 可以參考文檔進行搭建, 需要`docker-compose`.

**Comment by skq1998** [comment]
分享两个网站供大家把玩
- https://www.ejmastnak.com/tutorials/vim-latex/intro.html
- https://castel.dev/

**Comment by FlyCloudC** [comment]
TeXmacs比起来怎样？
https://www.yinwang.org/blog-cn/2012/09/18/texmacs

**Comment by audreyeternal** [comment]
真的建议overleaf，上大学的时候折腾环境花了不少时间，overleaf能跳过此环节。此外和zotero结合很方便写文章

**Reply by yliu-fe** [reply]
直接联动Zotero是Overleaf Premium才有的权限，一年90刀有点小贵。

**Comment by Cpt11** [comment]
[推荐一个机场，可以试用，不贵](https://bafang.vip/#/register?code=0mobpNZJ)

**Reply by CS-Trekker** [reply]
不需要，我都是白嫖的

**Comment by zhangeecs** [comment]
想问问大家，用LaTeX写毕业论文合适吗？这个好像不支持中文，大家是怎么解决的？先提前谢谢了

**Reply by TimothyWang-Tech** [reply]
已经毕业了叭哈哈，支持中文呀，用xelatex编译

**Comment by OsbertWang** [comment]
感谢推荐《一份简短的安装 LaTeX 的介绍》。在手册中，我也写到，国内用户，如果访问外网受限，可以考虑 TeXPage 网站，书中提供了链接。

**Comment by aaronleung1993** [comment]
https://www.learnlatex.org/zh-hans 好像这里有一个不错的LaTeX学习网站

**Comment by Elkeid-me** [comment]
Overleaf 免费版的编译时间只有 20 秒，大文档可能超时。

推荐以下两个平台：

- 国内的 [TeXPage](https://www.texpage.com/)，免费版给 1 分钟编译时间。
- 日本的 [Cloud LaTeX](https://cloudlatex.io/)，免费版给 2 分钟编译时间。

当然，直接在本地装一个 TeX 发行版是更好的选择。觉得 TeX Live 太大的可以考虑 MiKTeX 或 TinyTeX。

**Comment by zzzfreeuser** [comment]
有没有读文献的工具

**Reply by hvieacl** [reply]
Zotero挺不错的

**Comment by decetion** [comment]
现在overleaf只有20s免费编译时长了，可以说不交钱就用不了了，我当时还不知道作者的这本书，自己网上到处看教程磕磕碰碰用vscode+texlive本地部署了。
本地部署感觉也还好，辛苦几小时轻松好久好久~建议可以试试本地部署

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287

**Comment by L20wang** [comment]
至今无法理解为什么LATEX的环境配置如此抽象，你永远无法预测下一台配置LATEX的电脑能否正常编译
